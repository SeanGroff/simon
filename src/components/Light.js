// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  color: string,
  clickable: boolean,
  topLeft?: boolean,
  topRight?: boolean,
  bottomLeft?: boolean,
  bottomRight?: boolean,
};

const StyledLight = styled.div`
  display: inline-block;
  width: 200px;
  height: 200px;
  background-color: ${(props: Props) => props.color};
  border: 12px solid #333333;
  border-top-left-radius: ${(props: Props) => (props.topLeft ? '100%' : '0')};
  border-top-right-radius: ${(props: Props) => (props.topRight ? '100%' : '0')};
  border-bottom-left-radius: ${(props: Props) =>
    props.bottomLeft ? '100%' : '0'};
  border-bottom-right-radius: ${(props: Props) =>
    props.bottomRight ? '100%' : '0'};
  pointer-events: ${(props: Props) => (props.clickable ? 'auto' : 'none')};
`;

const Light = (props: Props) => <StyledLight {...props} />;

Light.defaultProps = {
  clickable: false,
};

export default Light;
