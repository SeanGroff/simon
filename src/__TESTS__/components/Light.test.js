import React from 'react';
import { shallow } from 'enzyme';
import Light from '../../components/Light';

it('should render a Light with a green background', () => {
  const component = shallow(<Light color="green" />);
  expect(component).toMatchSnapshot();
});
