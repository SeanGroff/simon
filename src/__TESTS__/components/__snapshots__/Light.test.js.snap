// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Light with a green background 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    clickable={false}
    color="green"
/>,
  "nodes": Array [
    <styled.div
      clickable={false}
      color="green"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Light
        clickable={false}
        color="green"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "clickable": false,
          "color": "green",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          clickable={false}
          color="green"
/>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          clickable={false}
          color="green"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Light
    clickable={false}
    color="green"
/>,
}
`;
